aiven/client/client.py

```diff
@@ -2750,9 +2750,18 @@ def byoc_update(
     def byoc_list(self, *, organization_id: str) -> Mapping[Any, Any]:
         return self.verify(self.get, self.build_path("organization", organization_id, "custom-cloud-environments"))
 
-    def byoc_provision(self, *, organization_id: str, byoc_id: str, aws_iam_role_arn: str | None) -> Mapping[Any, Any]:
+    def byoc_provision(
+        self,
+        *,
+        organization_id: str,
+        byoc_id: str,
+        aws_iam_role_arn: str | None = None,
+        google_privilege_bearing_service_account_id: str | None = None,
+    ) -> Mapping[Any, Any]:
         if aws_iam_role_arn is not None:
```

Comment:
if both `aws_iam_role_arn` and `google_privilege_bearing_service_account_id` are used, we will try to provision AWS byoc. This behaviour might get unpredictable once we support even more cloud providers

Can we add a check before to make sure there exactly one of these IAM role like arguments? what do you think?